test/js/specs/happy.js

```diff
@@ -2,125 +2,424 @@ const path = require('path')
+      input.sendKeys(path.join(__dirname, documentUpload.assignDir('passport.jpg')))
+      waitForUploadToFinish
+      const checkReadabilityText = documentUpload.title.getText()
+      expect(checkReadabilityText).to.equal(documentUploadLocale["confirm"]["document"]["title"])
+      const makeSureClearDetailsMessage = documentUploadConfirmation.makeSureClearDetailsMessage.getText()
+      expect(makeSureClearDetailsMessage).to.equal(documentUploadLocale["confirm"]["passport"]["message"])
+      documentUploadConfirmation.makeSureClearDetailsMessage.isDisplayed()
+    })
+
+    it('should upload driving licence and verify UI elements', async () => {
       driver.get(localhostUrl)
-      welcome.getPrimaryBtn().click()
-      documentSelection.getPassport().click()
+      welcome.primaryBtn.click()
+      documentSelection.drivingLicenceIcon.click()
+      const frontOfDrivingLicenceTitle = documentUpload.title.getText()
+      documentUpload.title.isDisplayed()
+      expect(frontOfDrivingLicenceTitle).to.equal(documentUploadLocale["capture"]["driving_licence"]["front"]["title"])
+      const frontOfDrivingLicenceInstructionMessage = documentUpload.uploaderInstructionsMessage.getText()
+      expect(frontOfDrivingLicenceInstructionMessage).to.equal(documentUploadLocale["capture"]["driving_licence"]["front"]["instructions"])
+      documentUpload.uploaderInstructionsMessage.isDisplayed()
+      const uploadFront = documentUpload.getUploadInput()
+      uploadFront.sendKeys(path.join(__dirname, documentUpload.assignDir('uk_driving_licence.png')))
+      waitForUploadToFinish
+      documentUploadConfirmation.confirmBtn.click()
+      const backOfDrivingLicenceTitle = documentUpload.title.getText()
+      expect(backOfDrivingLicenceTitle).to.equal(documentUploadLocale["capture"]["driving_licence"]["back"]["title"])
+      const backOfDrivingLicenceInstructionMessage = documentUpload.uploaderInstructionsMessage.getText()
+      expect(backOfDrivingLicenceInstructionMessage).to.equal(documentUploadLocale["capture"]["driving_licence"]["back"]["instructions"])
+      documentUpload.uploaderInstructionsMessage.isDisplayed()
+      const uploadBack = documentUpload.getUploadInput()
+      uploadBack.sendKeys(path.join(__dirname,documentUpload.assignDir('back_driving_licence.jpg')))
+      waitForUploadToFinish
+      const checkReadabilityText = documentUpload.title.getText()
+      expect(checkReadabilityText).to.equal(documentUploadLocale["confirm"]["document"]["title"])
+      documentUpload.title.isDisplayed()
+      const makeSureClearDetailsMessage = documentUploadConfirmation.makeSureClearDetailsMessage.getText()
+      expect(makeSureClearDetailsMessage).to.equal(documentUploadConfirmationLocale["confirm"]["driving_licence"]["message"])
+      documentUploadConfirmation.makeSureClearDetailsMessage.isDisplayed()
+    })
+
+    it('should upload identity card and verify UI elements', async () => {
```

Comment:
There is a lot of repetition between this scenario and the one above. You could avoid it by having an array or object where you map the doc type and icons for front and back. Then you could do something like this: 
_(rough example)_

```
driver.get(localhostUrl)
welcome.primaryBtn.click()
documentSelection[doc_type].click()
const title = documentUpload.title.getText()

expect(title).to.equal(documentUploadLocale["capture"][doc_type]["front"]["title"])
const instructions = documentUpload.uploaderInstructionsMessage.getText()
expect(instructions).to.equal(documentUploadLocale["capture"][doc_type]["front"]["instructions"])

const uploadFront = documentUpload.getUploadInput()
uploadFront.sendKeys(path.join(__dirname, documentUpload.assignDir(`${doc_type.icon.name}.png`)))
waitForUploadToFinish
documentUploadConfirmation.confirmBtn.click()

//if doc_type has back
const back_title = documentUpload.title.getText()
expect(back_title).to.equal(documentUploadLocale["capture"][doc_type]["back"]["title"])
const back_instructions = documentUpload.uploaderInstructionsMessage.getText()
expect(back_instructions).to.equal(documentUploadLocale["capture"][doc_type]["back"]["instructions"])

const uploadBack = documentUpload.getUploadInput()
uploadBack.sendKeys(path.join(__dirname,documentUpload.assignDir(`${doc_type.back.icon}.jpg`)))
waitForUploadToFinish
const checkReadabilityText = documentUpload.title.getText()
expect(checkReadabilityText).to.equal(documentUploadLocale["confirm"]["document"]["title"])
```
You could also wrap each section of this big scenario into a named function and make it more readable and reusable

Reply:
For 
`uploadFront.sendKeys(path.join(__dirname, documentUpload.assignDir(`${doc_type.icon.name}.png`)))`

I have already created a neater function in 1d01382

Reply:
Do you suggest we should have an object e.g.

`var object = {
      doc_type: ["driving_licence", "national_identity_card"],
      side: ["front", "back"]
    }`

and in the test step I should use e.g.
`expect(frontOfDrivingLicenceTitle).to.equal(documentUploadLocale["capture"][object.doc_type[0]][object.side[0]]["title"])`

instead of:

`expect(backOfDrivingLicenceTitle).to.equal(documentUploadLocale["capture"]["driving_licence"]["front"]["title"])` ??

Tbh, to me, it makes the test a bit less readable.

Reply:
You could make it a map like this (for example)
```js
var docTypeToSides = {
    driving_licence: {
        front: true,
        back: true
    },
    national_identity_card: {
        front: true,
        back: true
    },
    passport: {
        front: true,
        back: false
    }
}
```
so your test step it would look readable still

Reply:
Thanks Kimmi. I think since we talked about it, the readability of tests improved a lot so I'm not 100% how valuable it would be to apply it now.